autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -146,32 +151,35 @@ def transform(
 
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
             X = typing.cast(pd.DataFrame, X)
-            if np.any(pd.isnull(X)):
-                for column in X.columns:
-                    if X[column].isna().all():
-                        X[column] = pd.to_numeric(X[column])
+            # If we had null columns in our fit call and we made them numeric, then:
+            # - If the columns are null even in transform, apply the same procedure.
+            # - Otherwise, substitute the values with np.NaN and then make the columns numeric.
+            # If the column is null here, but it was not in fit, it does not matter.
+            for column in self.null_columns:
+                # The column is not null, make it null since it was null in fit.
+                if not X[column].isna().all():
+                    X[column] = np.NaN
+                X[column] = pd.to_numeric(X[column])
+
+            # for the test set, if we have columns with only null values
+            # they will probably have a numeric type. If these columns were not
+            # with only null values in the train set, they should be converted
+            # to the type that they had during fitting.
```

Comment:
How do we know if `X` is a training dataset or a test dataset?

Reply:
we dont. It means that the columns which have all null values in the test set it should either be numeric or some type that was the type of the column in the train set. So we are just converting this. maybe we can reduce this as it may be repeated for training and testing. So we can use `self._is_fitted` to check if its a test value or a train value